### PR TITLE
ensure that a config directory exists before trying to use it

### LINKED ISSUE #116: Should ensure that `config_dir() + "bindle"` exists
Right now, Bindle doesn't check to see if the dir exists before attempting to operate on it. While this is fine for reads, for writes it is problematic. See `create-key` for an example.

### PR DESCRIPTION
**Pull request by technosophos**
Make sure that the config dir exists before attempting to use it.

Closes #116

Signed-off-by: Matt Butcher <matt.butcher@microsoft.com>

### CODE REVIEW

File: bin/client/main.rs [hunk 1]
```diff
@@ -100,11 +100,11 @@ async fn main() -> std::result::Result<(), ClientError> {
                 SignatureRole::Creator
             };
             // Keyfile
-            let keyfile = sign_opts.secret_file.unwrap_or_else(|| {
-                default_config_dir()
-                    .unwrap_or_else(|| PathBuf::from("."))
-                    .join("secret_keys.toml")
-            });
+            let keyfile = match sign_opts.secret_file {
+                Some(dir) => dir,
```
Comment (thomastaylor312) [inline bin/client/main.rs:104]: Should you also ensure that this dir is created?

File: bin/client/main.rs [hunk 2]
```diff
@@ -142,11 +142,10 @@ async fn main() -> std::result::Result<(), ClientError> {
             println!("{}", toml::to_string_pretty(&label)?);
         }
         SubCommand::CreateKey(create_opts) => {
-            let dir = create_opts.secret_file.unwrap_or_else(|| {
-                default_config_dir()
-                    .unwrap_or_else(|| PathBuf::from("."))
-                    .join("secret_keys.toml")
-            });
+            let dir = match create_opts.secret_file {
+                Some(dir) => dir,
```
Comment (thomastaylor312) [inline bin/client/main.rs:146]: Same question here

**Review by thomastaylor312 — COMMENTED** [review]
One concern to discuss around ensuring the dir passed in by the user

File: bin/client/main.rs (hunk 2 above)
Comment (technosophos) [reply bin/client/main.rs:146]: Same answer. ;-)

File: bin/client/main.rs (hunk 1 above)
Comment (technosophos) [reply bin/client/main.rs:104]: I don't think so. My take is that if the user supplies a path, it is up to them to create the path. This is the way most programs work: If they use an `$XDG_*` dir, the program takes responsibility for creating (same for a hidden dir in `$HOME`). But if the user supplies a path, the tool assumes the user did the work.

In this particular case, per the Rust recommendations, we don't use `metadata()` and instead let the call to create the file be the point of failure.

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**